R/zchunk_LA143.HDDCDD.R

```diff
@@ -31,65 +30,184 @@ module_energy_LA143.HDDCDD_DISABLED <- function(command, ...) {
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     GIS_ctry <- get_data(all_data, "energy/GIS_ctry")
-    L101.Pop_thous_GCAM3_ctry_Y <- get_data(all_data, "L101.Pop_thous_GCAM3_ctry_Y")
-    i <- get_data(all_data, "i")
+    L101.Pop_thous_GCAM3_ctry_Y <- get_data(all_data, "L101.Pop_thous_GCAM3_ctry_Y") %>%
+      rename(population = value) %>% mutate(year = as.integer(year))
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # 1b. reading HDDCDD files in a loop as we do not know their names or how many there are
+    GISfilepath <- "./inst/extdata/energy/GIS/"
+    GISfiles <- list.files( GISfilepath )
+    GISfiles.list <- list()
+
+    # Identifying and removing base year (2010) data
+    base_year_HDDCDD_data <-  grep( "_2010", GISfiles )
+    GISfiles <- GISfiles[-base_year_HDDCDD_data]
+
+    # Putting all data into list
+    for( i in GISfiles){
+      index <- which( GISfiles == i )
+      GISfiles.list[[index]] <- read.csv( paste0(GISfilepath, i), stringsAsFactors = F)
+      GISfiles.list[[index]]$file <- sub(".csv","",i)
+    }
+
+    # Combining all HDDCDD files into one tibble
+    HDDCDD_data <- do.call( bind_rows, GISfiles.list ) %>%
+      tibble::as_tibble()
+
+    #--------------------------------------------------------------------------
+
+    #Currently the HDDCDD data stops at 2099. If this is the case, add 2100
+    if( !"X2100" %in% names( HDDCDD_data ) ){
```

Comment:
This is one of those things that always seems clunky in long form. Is there an easier way to do this than:
`  if ( ! 2100 %in% HDDCDD_data$year){`
    `data_2100 <- HDDCDD_data %>%`
    `filter(year == 2099) %>%`
   `mutate(year = as.integer(2100))`
    `HDDCDD_data <- HDDCDD_data %>% bind_rows(data_2100)
    }`